File: python/lib/helpers/runtime.py
from timeit import default_timer


def print_answer_and_elapsed_time(function):
    start = default_timer()
    answer = function()
    elapsed = 1000 * (default_timer() - start)

    print('Answer: %(answer)s' % locals())
    print('Elapsed time: %(elapsed)s ms' % locals())

File: haskell/src/Helpers/Runtime.hs
module Helpers.Runtime
( printAnswerAndElapsedTime
) where

import Control.Exception (evaluate)
import System.CPUTime (getCPUTime)
import Text.Printf (printf)

printAnswerAndElapsedTime :: (Show a) => a -> IO ()
printAnswerAndElapsedTime answer = do
    start <- getCPUTime
    value <- evaluate answer
    end <- getCPUTime
    printf "Answer: %s\n" $ show value
    printf "Elapsed time: %fms\n" $ fromIntegral (end - start) / (10 ^ 9)
